File: backend-csharp/DTOs/Responses/ProfileResponse.cs
namespace Picture2Text.Api.DTOs.Responses;

public class ProfileResponse
{
    public int Code { get; set; } = 200;
    public string Message { get; set; } = "操作成功";
    public ProfileData? Data { get; set; }
}

public class ProfileData
{
    public int Id { get; set; }
    public string IdNo { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
}

File: backend-nestjs/src/dto/response/profile.response.ts
import { ApiProperty } from '@nestjs/swagger';

export class ProfileData {
  @ApiProperty({
    description: '使用者 ID',
    example: 1,
  })
  Id: number;

  @ApiProperty({
    description: '使用者身分證號/員工編號',
    example: 'A123456789',
  })
  IdNo: string;

  @ApiProperty({
    description: '使用者姓名',
    example: '張三',
  })
  Name: string;
}

export class ProfileResponse {
  @ApiProperty()
  Code: number;

  @ApiProperty()
  Message: string;

  @ApiProperty({ type: ProfileData })
  Data: ProfileData;
}
